docs/naming_conventions.md

```diff
@@ -0,0 +1,71 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+1. A resource in the FHIR model should always have the `id` attribute
+populated by model developers. We should never let the server auto populate
+`id` since it is the primary way that machines and humans will remember,
+reference, and lookup a resource.
+
+2. The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
+
+3. All resource `id` values should have the same casing. There does not seem
+   to be a consistent standard or recommendation for this but we should pick
+   one and stick to it. Based on observations, the options are:
```

Comment:
Names for multiword identifier naming conventions are not standardized. The terms "camel case" and "snake case" ambiguously refer to several distinct casing strategies where, among other details, capitalization of the first letter is unspecified. I think you do need split out the different variants of each in order to pick one:

UpperCamelCase
lowerCamelCase
lowercase_with_underscores
etc